drivers/sensor/adltc2990/adltc2990.c

```diff
@@ -0,0 +1,471 @@
+#define ADLTC2990_REG_V3_LSB            0x0B
+#define ADLTC2990_REG_V4_MSB            0x0C
+#define ADLTC2990_REG_V4_LSB            0x0D
+#define ADLTC2990_REG_VCC_MSB           0x0E
+#define ADLTC2990_REG_VCC_LSB           0x0F
+
+#define VOLTAGE_SINGLE_ENDED_VALUES 2
+#define VOLTAGE_DIFF_VALUES         1
+#define TEMP_VALUES                 1
+#define CURRENT_VALUES              1
+#define MICROOHM_CONVERSION_FACTOR  1000000
+static int adltc2990_bus_reg_read_i2c(const struct device *dev, uint8_t reg_address, uint8_t *value)
+{
+	const struct adltc2990_config *cfg = dev->config;
+
+	return i2c_reg_read_byte_dt(&cfg->bus, reg_address, value);
+}
+
+static int adltc2990_bus_reg_write_i2c(const struct device *dev, uint8_t reg_address, uint8_t value)
+{
+	const struct adltc2990_config *cfg = dev->config;
+
+	return i2c_reg_write_byte_dt(&cfg->bus, reg_address, value);
+}
+
+static uint8_t adltc2990_get_controle_mode(uint8_t mode_v1_v2, uint8_t mode_v3_v4)
+{
+	uint8_t conf_value = 0xff;
+
+	if ((mode_v1_v2 == voltage_single_ended) &&
+	    (mode_v3_v4 == voltage_differential || mode_v3_v4 == current)) {
+		LOG_ERR("This configuration is not allowed");
+	} else if (mode_v1_v2 == nothing && mode_v3_v4 == nothing) {
+		conf_value = 0;
+	} else if (mode_v1_v2 == nothing &&
+		   (mode_v3_v4 == current || mode_v3_v4 == voltage_differential)) {
+		conf_value = 20;
+	} else if (mode_v1_v2 == nothing && mode_v3_v4 == voltage_single_ended) {
+		conf_value = 18;
+	} else if (mode_v1_v2 == nothing && mode_v3_v4 == temperature) {
+		conf_value = 16;
+	} else if ((mode_v1_v2 == current || mode_v1_v2 == voltage_differential) &&
+		   mode_v3_v4 == nothing) {
+		conf_value = 9;
+	} else if ((mode_v1_v2 == current || mode_v1_v2 == voltage_differential) &&
+		   (mode_v3_v4 == current || mode_v3_v4 == voltage_differential)) {
+		conf_value = 30;
+	} else if ((mode_v1_v2 == current || mode_v1_v2 == voltage_differential) &&
+		   mode_v3_v4 == voltage_single_ended) {
+		conf_value = 26;
+	} else if ((mode_v1_v2 == current || mode_v1_v2 == voltage_differential) &&
+		   mode_v3_v4 == temperature) {
+		conf_value = 25;
+	} else if (mode_v1_v2 == voltage_single_ended && mode_v3_v4 == nothing) {
+		conf_value = 8;
+	} else if (mode_v1_v2 == voltage_single_ended &&
+		   (mode_v3_v4 == current || mode_v3_v4 == voltage_differential)) {
+	} else if (mode_v1_v2 == voltage_single_ended && mode_v3_v4 == voltage_single_ended) {
+		conf_value = 31;
+	} else if (mode_v1_v2 == voltage_single_ended && mode_v3_v4 == temperature) {
+		conf_value = 24;
+	} else if (mode_v1_v2 == temperature && mode_v3_v4 == nothing) {
+		conf_value = 12;
+	} else if (mode_v1_v2 == temperature &&
+		   (mode_v3_v4 == current || mode_v3_v4 == voltage_differential)) {
+		conf_value = 28;
+	} else if (mode_v1_v2 == temperature && mode_v3_v4 == voltage_single_ended) {
+		conf_value = 11;
+	} else if (mode_v1_v2 == temperature && mode_v3_v4 == temperature) {
+		conf_value = 29;
+	}
```

Comment:
There are way too many `else if` clauses; please try to simplify

Reply:
done